src/views/sources/addSourceModal.tsx

```diff
@@ -42,33 +43,18 @@ const AddSourceModal: React.FC<AddSourceModalProps> = ({ source, type, onClose,
     source?.options?.disable_ssl ? 'Disable SSL' : source?.options?.ssl_protocol || 'SSLv23'
   );
 
-  const typeValue = source?.source_type || type.split(' ').shift()?.toLowerCase();
-  const isNetwork = typeValue === 'network';
+  const sourceTypeValue = source?.source_type || sourceType.split(' ').shift()?.toLowerCase();
```

Comment:
required: update this line to allow optional chaining, this will allow the optional/undefined value

`const sourceTypeValue = source?.source_type || sourceType?.split(' ')?.shift()?.toLowerCase();`